tests/hkdf.cc

```diff
@@ -0,0 +1,239 @@
+/*
+ *    Copyright 2022 Cryspen Sarl
+ *
+ *    Licensed under the Apache License, Version 2.0 or MIT.
+ *    - http://www.apache.org/licenses/LICENSE-2.0
+ *    - http://opensource.org/licenses/MIT
+ */
+
+#include <fstream>
+
+#include <gtest/gtest.h>
+#include <nlohmann/json.hpp>
+
+#include "Hacl_HKDF.h"
+#include "Hacl_Spec.h"
+#include "util.h"
+
+using json = nlohmann::json;
+
+typedef struct
+{
+  std::string comment;
+  // A tcId of 0 indicates "not Wycheproof".
+  uint32_t tcId;
+  std::string hash;
+  // 1) Input Keying Material
+  bytes IKM;
+  bytes salt;
+  bytes info;
+  // Length of OKM
+  uint32_t L;
+  // 2) A pseudorandom key (of HashLen octets)
+  // Only available when "not Wycheproof".
+  bytes PRK;
+  // 3) Output Keying Material
+  bytes OKM;
+  bool valid;
+} TestCase;
+
+class HKDFSuite : public ::testing::TestWithParam<TestCase>
+{};
+
+TEST_P(HKDFSuite, TestCase)
+{
+  auto test = GetParam();
+  std::cout << "# " << test.comment << std::endl;
+
+  // TODO: How does error handling in Hacl_HKDF_* work?
```

Comment:
Either remove the todo or file an issue and reference it here.

Reply:
I opened https://github.com/cryspen/hacl-packages/issues/123.